### PR TITLE
Add doc section about doobie integration

### LINKED ISSUE #137: Docs: Doobie section
We need to document the Doobie integration and how you can easily interleave queries in FreeS programs with a few examples.

### PR DESCRIPTION
**Pull request by peterneyens**
Fixes #137.

### CODE REVIEW

File: docs/src/main/tut/docs/integrations/doobie.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    def subtract(a: Int, b: Int): Task[Int] = Task.now(a - b)
+  }
+```
+
+A program using the `Example` module:
+
+```tut:book
+def example[F[_]: DoobieM](implicit example: Example[F]): FreeS[F, (Person, Int)] =
+  for {
+    person <- getPerson.liftFS[F]  // analogous to example.doobieM.transact(getPerson)
```
Comment (peterneyens) [inline docs/src/main/tut/docs/integrations/doobie.md:99]: To use the lifting syntax here, I needed to add the `DoobieM` context bound even though we have an implicit `Example`.

Should we look if we can provide implicit conversions from modules to the underlying algebras, so the syntax can be used directly?

Comment (raulraja) [reply docs/src/main/tut/docs/integrations/doobie.md:99]: That would be awesome!

**Review by raulraja — APPROVED** [review]

**Comment by raulraja** [comment]
Looks great @peterneyens, I think we need to fix the link in the menu though to point to these docs.

**Comment by peterneyens** [comment]
I have no idea how the test coverage can go down with this PR ?

**Comment by raulraja** [comment]
@peterneyens https://docs.codecov.io/docs/unexpected-coverage-changes They explain why this may be happening but definitely nothing in this PR changing coverage

**Pull request merged**